types/parameterized_types.go

```diff
@@ -0,0 +1,112 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+// IntegerParam represents a single integer parameter for a parameterized type
+// Example: VARCHAR(L1) -> L1 is the integer parameter
+type IntegerParam struct {
+	Name string
+}
+
+func (m IntegerParam) Equals(o IntegerParam) bool {
+	return m == o
+}
+
+func (p IntegerParam) ToProto() *proto.ParameterizedType_IntegerParameter {
+	panic("not implemented")
+}
```

Comment:
same comment as before

Reply:
removed it since now Type interface doesn't have proto methods